### PR TITLE
chore: proposal to deploy v2.2.0

### PR DESCRIPTION
**Pull request by gregorydemay**

### CODE REVIEW

File: proposals/evm_rpc_2024_10_18.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    * See this [forum post](https://forum.dfinity.org/t/evm-rpc-canister/23313/53) for more information.
+* Support a new threshold strategy to aggregate responses from multiple providers. This allows callers to be more robust against a single provider failure.
+* Added a new convenience method for `eth_call` to allow calling read-only functions of a smart contract.
+
+## Install args
+
+```
+git fetch
+git checkout 3de8830e509cb97411367a153494e524573903f4
+didc encode -d evm_rpc.did -t '(InstallArgs)' '(record {demo = opt false; manageApiKeys = opt vec {principal "mf7xa-laaaa-aaaar-qaaaa-cai"}; logFilter = null})' | xxd -r -p | sha256sum
```
Comment (gregorydemay) [inline proposals/evm_rpc_2024_10_18.md:31]: `4449444c066c03dcc690fd0301a3b299930403c98fb9b104046e026b0484809ec7047fdf92a2a4087fb3f594920d718e9bd6980f716e7e6e056d6801000001000101010a00000000023000000101`

Comment (ninegua) [reply proposals/evm_rpc_2024_10_18.md:31]: 4449444c066c03dcc690fd0301a3b299930403c98fb9b104046e026b0484809ec7047fdf92a2a4087fb3f594920d718e9bd6980f716e7e6e056d6801000001000101010a00000000023000000101

But there are two problems:
1. it should be `candid/evm_rpc.did`.
2. The above hex is the output before `xxd` command. So line 31 is also doing sha256sum, not sure if this is what you want.

File: proposals/evm_rpc_2024_10_18.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+# Proposal to reinstall the EVM RPC canister
+
+Repository: `https://github.com/internet-computer-protocol/evm-rpc-canister.git`
+
+Git hash: `3de8830e509cb97411367a153494e524573903f4`
+
+New compressed Wasm hash: `636442e4349a007316dcc1452d9a256d22203186181d2534f33bd264162305b3`
```
Comment (ninegua) [inline proposals/evm_rpc_2024_10_18.md:7]: Wasm hash checked

**Review by ninegua — APPROVED** [review]
Does it require manually setting API keys in order for the EVM canister to function once this is deployed?

File: proposals/evm_rpc_2024_10_18.md (hunk 1 above)
Comment (gregorydemay) [reply proposals/evm_rpc_2024_10_18.md:31]: Thanks @ninegua for checking!
> it should be candid/evm_rpc.did.

Absolutely right, good catch!

> The above hex is the output before xxd command. So line 31 is also doing sha256sum, not sure if this is what you want.

Actually yes, with the new proposal types only the hash of the args is displayed, see this [thread](https://dfinity.slack.com/archives/C0175CHJ0P6/p1723237367803209)

**Comment by gregorydemay** [comment]
> Does it require manually setting API keys in order for the EVM canister to function once this is deployed?

@ninegua : Yes! I will prepare a script locally

File: proposals/evm_rpc_2024_10_18.md (hunk 1 above)
Comment (THLO) [reply proposals/evm_rpc_2024_10_18.md]: I don't get the same hash. I also stumbled over candid/evm_rpc.did. After fixing this, I get

```
$ didc encode -d candid/evm_rpc.did -t '(InstallArgs)' '(record {demo = opt false; manageApiKeys = opt vec {principal "mf7xa-laaaa-aaaar-qaaaa-cai"}; logFilter = null})' | xxd -r -p | sha256sum
3d28b9b62643bc7875c5c611b13bdfb39aef6f342fa8af74ad0a5e8998003f31

$ git status
HEAD detached at 3de8830
```

Comment (gregorydemay) [reply proposals/evm_rpc_2024_10_18.md]: @THLO Sorry for the confusion,

> 3d28b9b62643bc7875c5c611b13bdfb39aef6f342fa8af74ad0a5e8998003f31

is the correct hash of

> 4449444c066c03dcc690fd0301a3b299930403c98fb9b104046e026b0484809ec7047fdf92a2a4087fb3f594920d718e9bd6980f716e7e6e056d6801000001000101010a00000000023000000101

so everything is good ✅ . To double check you can do

```
echo "4449444c066c03dcc690fd0301a3b299930403c98fb9b104046e026b0484809ec7047fdf92a2a4087fb3f594920d718e9bd6980f716e7e6e056d6801000001000101010a00000000023000000101" | xxd -r -p | sha256sum
```

File: proposals/evm_rpc_2024_10_18.md (hunk 2 above)
Comment (THLO) [reply proposals/evm_rpc_2024_10_18.md]: Verified!

**Review by rvanasa — APPROVED** [review]

**Pull request merged**